Dockerfile

```diff
@@ -69,12 +79,38 @@ ENV JAVA_DEBIAN_VERSION=unused
 ENV JAVA_VERSION=unused
 
 # Setup environment variables to point to all jvms we have
-ENV JAVA_7_HOME=/usr/lib/jvm/zulu7
-ENV JAVA_8_HOME=/usr/lib/jvm/openjdk8
-ENV JAVA_11_HOME=/usr/lib/jvm/openjdk11
-ENV JAVA_17_HOME=/usr/lib/jvm/openjdk17
+ENV JAVA_8_HOME=/usr/lib/jvm/8
+ENV JAVA_11_HOME=/usr/lib/jvm/11
+ENV JAVA_17_HOME=/usr/lib/jvm/17
+
+ENV JAVA_HOME=${JAVA_8_HOME}
+ENV PATH=${JAVA_HOME}/bin:${PATH}
+
+FROM base AS variant
+ARG VARIANT_LOWER
+ARG VARIANT_UPPER
 
-ENV JAVA_ZULU7_HOME=/usr/lib/jvm/zulu7
+COPY --from=all-jdk /usr/lib/jvm/${VARIANT_LOWER} /usr/lib/jvm/${VARIANT_LOWER}
+ENV JAVA_${VARIANT_UPPER}_HOME=/usr/lib/jvm/${VARIANT_LOWER}
+ENV JAVA_${VARIANT_LOWER}_HOME=/usr/lib/jvm/${VARIANT_LOWER}
+
+# Full image for debugging, contains all JDKs.
+FROM base AS full
+
+COPY --from=all-jdk /usr/lib/jvm/7 /usr/lib/jvm/7
+COPY --from=all-jdk /usr/lib/jvm/zulu8 /usr/lib/jvm/zulu8
+COPY --from=all-jdk /usr/lib/jvm/zulu11 /usr/lib/jvm/zulu11
+COPY --from=all-jdk /usr/lib/jvm/oracle8 /usr/lib/jvm/oracle8
+COPY --from=all-jdk /usr/lib/jvm/ibm8 /usr/lib/jvm/ibm8
+COPY --from=all-jdk /usr/lib/jvm/semeru8 /usr/lib/jvm/semeru8
+COPY --from=all-jdk /usr/lib/jvm/semeru11 /usr/lib/jvm/semeru11
+COPY --from=all-jdk /usr/lib/jvm/semeru17 /usr/lib/jvm/semeru17
+COPY --from=all-jdk /usr/lib/jvm/graalvm11 /usr/lib/jvm/graalvm11
+COPY --from=all-jdk /usr/lib/jvm/graalvm17 /usr/lib/jvm/graalvm17
+
+ENV JAVA_7_HOME=/usr/lib/jvm/7
```

Comment:
Is Java_7_HOME required in `default-jdk` in order to run the `test-published-dependencies` tests?

Reply:
I have forced the `7` image for the 2 jobs that require Java 7: https://github.com/DataDog/dd-trace-java/pull/4877/files#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47R573

I avoided getting them in base image since most jobs don't need it.